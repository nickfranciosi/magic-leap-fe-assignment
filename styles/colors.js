const alert = '#e15021';
const backgroundMain = '#151515';
const border = '#333';
const callout = '#9e4f60';
const callToAction = '#edec51';
const subduedText = '#ddd';
const primaryText = '#fff';

export default {
  alert,
  backgroundMain,
  border,
  callout,
  callToAction,
  subduedText,
  primaryText
};
