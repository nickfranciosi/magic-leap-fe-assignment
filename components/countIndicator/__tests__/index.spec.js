/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import CountIndicator from '../index';
import colors from '../../../styles/colors';

describe('CountIndicator', () => {
  test('it should render', () => {
    const tree = renderer.create(<CountIndicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should be round', () => {
    const tree = renderer.create(<CountIndicator />).toJSON();
    expect(tree).toHaveStyleRule('border-radius', '50%');
  });

  test('it should have show legible text', () => {
    const tree = renderer.create(<CountIndicator />).toJSON();
    expect(tree).toHaveStyleRule('color', colors.primaryText);
    expect(tree).toHaveStyleRule('background-color', colors.alert);
  });
});
