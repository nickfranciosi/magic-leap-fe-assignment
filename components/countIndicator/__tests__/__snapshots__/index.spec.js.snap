// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountIndicator it should render 1`] = `
.c0 {
  background-color: #e15021;
  display: block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  color: #fff;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: -5px;
  right: -2px;
  font-size: 11px;
  font-family: 'Lato',Helvetica,arial,sans-serif;
}

<div
  className="c0"
/>
`;
